src/contracts/digital-twin/digital-twin.spec.ts

```diff
@@ -186,9 +258,261 @@ describe('DigitalTwin', function test() {
+          containerAddress = entries[entry].value.config.address;
+          containerContract = await localRuntime.contractLoader.loadContract('DataContract', containerAddress);
+          containerDescriptionHash = await localRuntime.executor.executeContractCall(
+            containerContract,
+            'contractDescription',
+          );
+          containerEntries = await (twin as any).getContainerEntryHashes( // sshhh
+            containerContract,
+            containerDescriptionHash,
+          );
+          pinnedHashes = pinnedHashes.concat(containerEntries);
+          pinnedHashes.push(await localRuntime.executor.executeContractCall(containerContract, 'sharing'));
+          pinnedHashes.push(containerDescriptionHash);
+        }
+      }
+      pinnedHashes.push(twinDescriptionHash);
+
+      const pinnedHashesAfterCreation = await getPinnedFileHashes();
+
+      // Make sure hashes have indeed been pinned, as a safety mechanism
+      // in case the smart agent acts weird
+      pinnedHashes.forEach((hash) => {
+        expect(
+          pinnedHashesAfterCreation.includes(hash),
+          'Hash should have been pinned upon creation, but wasn\'t',
+        ).to.be.true;
+      });
+
+      await twin.deactivate();
+
+      // Check if all containers' owner == 0x0
+      // & consumers are removed
+      // & authority is removed
+      // & description is unpinned
+      // & sharing is unpinned & removed
+      let containerOwner;
+      let containerAuthority;
+      let consumerCount;
+      let sharingAddress;
+      for (const entry of Object.keys(entries)) {
+        if (entries[entry].entryType === DigitalTwinEntryType.Container) {
+          containerAddress = entries[entry].value.config.address;
+          containerContract = await localRuntime.contractLoader.loadContract('DataContract', containerAddress);
+          containerOwner = await localRuntime.executor.executeContractCall(containerContract, 'owner');
+          containerAuthority = await localRuntime.executor.executeContractCall(containerContract, 'authority');
+          consumerCount = await localRuntime.executor.executeContractCall(containerContract, 'consumerCount');
+          sharingAddress = await localRuntime.executor.executeContractCall(containerContract, 'sharing');
+
+          expect(containerOwner).to.eq(nullAddress);
+          expect(containerAuthority).to.eq(nullAddress);
+          expect(consumerCount.toString()).to.eq('0'); // BigNumber
+          expect(sharingAddress).to.eq(nullBytes32);
+        }
+      }
+
+      // Check if twin's description is 0x0
+      // & twin's owner is 0x0
+      // & twin's authority is 0x0
+      // & twin's identity's owner is 0x0
+      const twinOwner = await localRuntime.executor.executeContractCall(
+        twinContract,
+        'owner',
+      );
+      const twinAuthority = await localRuntime.executor.executeContractCall(
+        twinContract,
+        'owner',
```

Comment:
owner checked twice? do you mean 'authority' ?
```suggestion
        'authority',
```